### PR TITLE
connect nav to redux +

### LINKED ISSUE #38: react-navigation + redux
`react-navigation` supports redux, but currently that's not implemented yet.

I just did this in a client project a few days ago, but it would be nice to land it in this boilerplate.

[Here's a gist](https://gist.github.com/skellock/fb15c65079bb973490cf7aa370f6fca9) with the juicy parts.  Next step is to roll this in by default.

### PR DESCRIPTION
**Pull request by GantMan**
This successfully connects our default navigation to redux out the box: 

Fixes #38

### CODE REVIEW

File: boilerplate/App/Redux/ScreenTrackingMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { NavigationActions } from 'react-navigation'
```
Comment (derekgreenberg) [inline boilerplate/App/Redux/ScreenTrackingMiddleware.js]: This is really cool.

File: boilerplate/App/Redux/index.js [hunk 2]
```diff
@@ -1,10 +1,17 @@
 import { combineReducers } from 'redux'
 import configureStore from './CreateStore'
 import rootSaga from '../Sagas/'
+import AppNavigation from '../Navigation/AppNavigation'
```
Comment (derekgreenberg) [inline boilerplate/App/Redux/index.js]: Suggestion: move the nav reducer to its own file and import it here. The nav reducer can grow considerably as it is tasked with handling navigation in response to Redux action dispatches.

Comment (GantMan) [inline boilerplate/App/Redux/index.js]: Good call!  I like that plan.

**Comment by GantMan** [comment]
Closed in favor of #85

**Pull request closed without merging**